limitador/src/storage/redis/redis_cached.rs

```diff
@@ -93,7 +94,7 @@ impl AsyncCounterStorage for CachedRedisStorage {
                             first_limited = Some(a);
                         }
                         if load_counters {
-                            counter.set_remaining(val);
+                            counter.set_remaining(counter.max_value() - val);
```

Comment:
Maybe we could get rid of this, no? Since we are unifying the way we apply limits, we won't keep track of remaining because of subtracting but checking if we reach the limit instead

Reply:
Which part? Setting that remainder is to set the headers in the response to the user, to inform them of how many request they still can perform in ... (see line below) the time they have left on their counter. 
There isn't much for us to decide, as that's all part of the [draft RFC](https://datatracker.ietf.org/doc/draft-ietf-httpapi-ratelimit-headers/)